### PR TITLE
Atlas Docs

### PR DESCRIPTION
**Pull request by bensojona**
Adding docs for Atlas. It appears that the dpl gem hasn't been updated with the latest Atlas provider, so I wasn't able to test, but I think this should get us close.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -354,6 +355,31 @@ It is possible to set file-specific `Cache-Control` and `Expires` headers using
...
+* `-include=<path>` (Optional): Glob pattern of files/directories to include (this may be specified multiple times, any excludes will override conflicting includes).
+* `-address=<url>` (Optional): The address of the Atlas server.
+* `-vcs` (Optional): Use VCS to determine which files to include/exclude.
+* `-metadata=<k=v>` (Optional): Arbitrary key-value (string) metadata to be sent with the upload; may be specified multiple times.
+* `-debug` (Optional): Turn on debug output.
+* `-version` (Optional): Print the version of this application.
+
+#### Examples:
+
+    dpl --provider=atlas --token=ATLAS_USERNAME --app=ATLAS_USERNAME/APP_NAME
```
Comment (KFishner) [inline README.md]: @bensojona the token flag should take ATLAS_TOKEN right?

File: README.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 Dpl supports the following providers:
 
 * [AppFog](#appfog)
+* [Atlas](#atlas)
```
Comment (KFishner) [inline README.md]: let's do Atlas by HashiCorp

File: README.md [hunk 3]
```diff
@@ -354,6 +355,31 @@ It is possible to set file-specific `Cache-Control` and `Expires` headers using
     dpl --provider=appfog --email=<email> --password=<password>
     dpl --provider=appfog --email=<email> --password=<password> --app=<app>
 
+### Atlas:
+
+The Atlas provider uses the [`atlas-upload-cli`](https://github.com/hashicorp/atlas-upload-cli) command. The [Atlas Upload CLI](https://github.com/hashicorp/atlas-upload-cli) is a lightweight command line interface for uploading application code to [Atlas](https://atlas.hashicorp.com/homepage?utm_source=github&utm_medium=travis-ci&utm_campaign=dpl) to kick off Atlas-based deployment processes from Travis CI.
+
+You first need to create an [Atlas account](https://atlas.hashicorp.com/account/new?utm_source=github&utm_medium=travis-ci&utm_campaign=dpl), then generate an [Atlas API token](https://atlas.hashicorp.com/settings/tokens) for Travis CI.
+
+#### Options:
```
Comment (meatballhat) [inline README.md]: These options should probably be in the same format as other providers where the option is shown as **bold-words** and without `=<arg>`.

File: README.md [hunk 4]
```diff
@@ -354,6 +355,31 @@ It is possible to set file-specific `Cache-Control` and `Expires` headers using
...
+* `--app` (Required): Atlas application name (`<atlas-username>/<app-name>`).
+* `-exclude=<path>` (Optional): Glob pattern of files or directories to exclude (this may be specified multiple times).
+* `-include=<path>` (Optional): Glob pattern of files/directories to include (this may be specified multiple times, any excludes will override conflicting includes).
+* `-address=<url>` (Optional): The address of the Atlas server.
+* `-vcs` (Optional): Use VCS to determine which files to include/exclude.
+* `-metadata=<k=v>` (Optional): Arbitrary key-value (string) metadata to be sent with the upload; may be specified multiple times.
+* `-debug` (Optional): Turn on debug output.
+* `-version` (Optional): Print the version of this application.
+
+#### Examples:
```
Comment (meatballhat) [inline README.md]: All of the options should be `--` (double dash) even though the generated command passed to the atlas cli contains the single dash form.

**Comment by meatballhat** [comment]
@bensojona Thanks for taking a crack at this!  The Atlas provider is now in master, although it's not in a published major release, afaict.

@BanzaiMan Do you know when we can expect the next `dpl` gem release?

File: README.md (hunk 3 above)
Comment (bensojona) [inline README.md]: @meatballhat ah yes, updated this.

File: README.md (hunk 4 above)
Comment (bensojona) [inline README.md]: @meatballhat I just ended up removing the dashes in the options list to mimic the other providers as well, no need for them if it's all consistent.

**Comment by bensojona** [comment]
@meatballhat thanks for your comments, let me know if anything else should be updated!

**Pull request merged**